### PR TITLE
Updated login error handler

### PR DESCRIPTION
**Pull request by nakel-ola**

### CODE REVIEW

File: app/api/auth/login/gallery/route.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  try {
+    await connectMongoDB();
+
+    const data = await request.json();
+
+    const { email, password } = data;
+
+    const user = await AccountGallery.findOne<GallerySchemaTypes>(
+      { email },
+      {
```
Comment (Moses-py) [inline app/api/auth/login/gallery/route.ts:21]: Please re-format this line to be:
const user = await AccountGallery.findOne<GallerySchemaTypes>({email}, "gallery_id, password"})

This is the recommendation from the mongoose documentation

File: app/api/auth/login/individual/route.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+export async function POST(request: Request) {
+  try {
+    await connectMongoDB();
+
+    const data = await request.json();
+
+    const { email, password } = data;
+
+    const user = await AccountIndividual.findOne<IndividualSchemaTypes>(
```
Comment (Moses-py) [inline app/api/auth/login/individual/route.ts:19]: Please re-format this line to be:
const user = await AccountGallery.findOne({email}, "user_id, password"})

This is the recommendation from the mongoose documentation

File: services/gallery-login.ts [hunk 3]
```diff
@@ -17,12 +17,11 @@ export const GalleryLoginProvider = CredentialsProvider({
 
       const data = await response.json();
 
-      return {
-        id: data._id,
-        ...data,
-      };
+      return { id: data.id };
     } catch (error) {
       console.log(error);
```
Comment (Moses-py) [inline services/gallery-login.ts:22]: Please use the error handler function to handle errors here. Thanks

File: app/api/auth/login/individual/route.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    const { user_id } = user;
+
+    return res.json({
+      status: 201,
+      message: "Login successfull",
+      id: user_id,
+    });
+  } catch (error) {
+    return handleErrorEdgeCases(error);
```
Comment (Moses-py) [inline app/api/auth/login/individual/route.ts]: The method for handling errors here has been updated. Please refer to the signup routes for reference.

The updates will be available after the next commit on my end

**Review by Moses-py — CHANGES_REQUESTED** [review]

File: app/api/auth/login/gallery/route.ts (hunk 1 above)
Comment (nakel-ola) [reply app/api/auth/login/gallery/route.ts:21]: oh, i didn't know you can do that

Comment (nakel-ola) [reply app/api/auth/login/gallery/route.ts:21]: can you provide the link

File: app/api/auth/login/individual/route.ts (hunk 4 above)
Comment (nakel-ola) [reply app/api/auth/login/individual/route.ts]: yes, i saw that

File: app/api/auth/login/gallery/route.ts (hunk 1 above)
Comment (Moses-py) [reply app/api/auth/login/gallery/route.ts:21]: mongoose

Comment (Moses-py) [reply app/api/auth/login/gallery/route.ts:21]: https://mongoosejs.com/docs/api/model.html#Model.findOne()

File: services/gallery-login.ts (hunk 3 above)
Comment (nakel-ola) [reply services/gallery-login.ts:22]: i will throw an error because if i return the  handleErrorEdgeCases typescript will show error

Comment (Moses-py) [reply services/gallery-login.ts:22]: Alright, please keep track of the behavior further down the link

**Review by Moses-py — APPROVED** [review]

**Review by Moses-py — APPROVED** [review]

**Pull request merged**